bin/tools.py

```diff
@@ -193,31 +193,49 @@ def get_pos(id):
 
         if config.args.order_from_ccs:
             # Sort by increasing difficulty, extracted from the CCS api.
-            # Get active contest.
+            class ProblemStat:
+                def __init__(self):
+                    self.solved = 0
+                    self.submissions = 0
+                    self.pending = 0
+                    self.teams_submitted = 0
+                    self.teams_pending = 0
+
+                def update(self, team_stats: dict[str, Any]):
+                    if team_stats["solved"]:
+                        self.solved += 1
+                    if team_stats["num_judged"]:
+                        self.submissions += team_stats["num_judged"]
+                        self.teams_submitted += 1
+                    if team_stats["num_pending"]:
+                        self.pending += team_stats["num_pending"]
+                        self.teams_pending += 1
+
+                def key(self, fall_back: str) -> tuple[int, float, str]:
+                    # self.solved more AC => easier
+                    # possible tie breakers:
+                    # self.submissions more needed to get the same number of AC => Harder
+                    # self.teams_pending more teams tried => appeared easier
```

Comment:
I don't see `team_pending` as part of the returned `key`?

Reply:
it is not used right now ^^ I think it is unlikely that `team_pending` or `submissions`  is equal. If we want to use both we most likely want to combine them in some way?